packages/jest-core/src/runJest.ts

```diff
@@ -292,6 +296,7 @@ export default async function runJest({
   const scheduler = await createTestScheduler(globalConfig, {
     startRun,
     ...testSchedulerContext,
+    runGlobalTeardown,
```

Comment:
why do we need to pass this in instead of just ensuring we call it here?

Reply:
Do you mean there's an easier way to access it from inside TestScheduler.ts?

![image](https://github.com/jestjs/jest/assets/1962469/adad9264-6f50-4153-8267-cde328ab903c)

Reply:
meant to run it from `runJest.ts`

Reply:
Just to clarify, we are talking about `private async _bailIfNeeded` which needs to be able to access globalTeardown callback.

Reply:
You call it like this

![image](https://github.com/jestjs/jest/assets/1962469/4e399119-d307-46ba-a5c3-bbce4f00c807)

Basically after each test you leave a window of opportunity to bail (i.e. process.exit())